### PR TITLE
Release ci

### PR DESCRIPTION
**Pull request by IdanKoblik**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation)
- [x] 🐞 Bug fix (fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)

### 🧪 How Has This Been Tested?
img

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/callable.test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          run: ./gradlew test
+        - name: Publish Test Report
+          uses: mikepenz/action-junit-report@v4
+          if: success() || failure()
+          with:
+           token: ${{ secrets.GITHUB_TOKEN }} 
+           detailed_summary: true
+           follow_symlink: true
+           check_name: |
+                commands
```
Comment (liorsl) [inline .github/workflows/callable.test.yml:34]: forgot something

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -4,67 +4,137 @@
...
+  workflow_dispatch:
+    inputs:
+      name:
+        description: 'Version'
+        required: true
+        type: string
 
 permissions:
-  contents: read
+  pull-requests: write
```
Comment (liorsl) [inline .github/workflows/release.yml:17]: why is this needed?

**Review by liorsl — CHANGES_REQUESTED** [review]

File: .github/workflows/callable.test.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: Run tests
+
+on:
+    workflow_call:
+        
+permissions:
+    contents: read
+    packages: write
+    id-token: write
+    pages: write
```
Comment (liorsl) [inline .github/workflows/callable.test.yml:10]: why is this needed?

File: .github/workflows/callable.test.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: Run tests
+
+on:
+    workflow_call:
+        
+permissions:
+    contents: read
+    packages: write
```
Comment (liorsl) [inline .github/workflows/callable.test.yml:8]: why is this needed?

File: .github/workflows/callable.test.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    id-token: write
```
Comment (liorsl) [inline .github/workflows/callable.test.yml:9]: why is this needed?

File: .github/workflows/callable.test.yml (hunk 1 above)
Comment (liorsl) [reply .github/workflows/callable.test.yml:36]: you didnt fix this

File: .github/workflows/callable.writerside.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+name: Run tests
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+  packages: write
```
Comment (liorsl) [inline .github/workflows/callable.writerside.yml:8]: why is this needed?

File: .github/workflows/callable.writerside.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  id-token: write
```
Comment (liorsl) [inline .github/workflows/callable.writerside.yml:9]: why is this needed?

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -3,11 +3,13 @@ name: Snapshot run
 
 on:
   push:
-    branches: ["main"]
+    branches: ["release-ci"]
 
 # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
 permissions:
-  contents: read
+  pull-requests: write
```
Comment (liorsl) [inline .github/workflows/main.yml:10]: why is this needed?

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -44,16 +47,60 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           APARTIUM_NEXUS_PASSWORD: ${{ secrets.APARTIUM_NEXUS_PASSWORD }}
           APARTIUM_NEXUS_USERNAME: ${{ secrets.APARTIUM_NEXUS_USERNAME }}
+
+      - name: Pre writerside
+        env:
+          new_version: snapshot
```
Comment (liorsl) [inline .github/workflows/main.yml:53]: Variables???

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -1,70 +1,297 @@
...
           distribution: 'adopt'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+      
+      - name: Pre writerside
+        env:
+          new_version: ${{ github.event.inputs.name }}
+        run: |
+          input_file="Writerside/writerside.cfg"
```
Comment (liorsl) [inline .github/workflows/release.yml:92]: looks familiar

**Review by liorsl — CHANGES_REQUESTED** [review]

File: .github/workflows/callable.test.yml (hunk 3 above)
Comment (ikfir) [reply .github/workflows/callable.test.yml:8]: why?

File: build-scripts/prepare-web-output.sh [hunk 11]
```diff
@@ -1,13 +1,14 @@
...
-mkdir -p docs/spigot
-mkdir -p docs/commands-spigot
-mkdir -p docs/commands
-ls docs/
-cp -a common/build/docs/javadoc/* docs/common
-cp -a spigot/build/docs/javadoc/* docs/spigot
-cp -a commands/build/docs/javadoc/* docs/commands
-cp -a commands/spigot-platform/build/docs/javadoc/* docs/commands-spigot
\ No newline at end of file
+mkdir $docs
```
Comment (ikfir) [inline build-scripts/prepare-web-output.sh:3]: add if to make sure `$docs` is not empty / null

File: Writerside/topics/Home.md [hunk 12]
```diff
@@ -1,8 +1,13 @@
-# Cocoa Beans
+# 🫘 Cocoa Beans
 
 Welcome to the cocoa-beans wiki!
 Cocoa beans is a toolkit for you to take with you to every new project. Instead of copying and pasting utilities code between projects, merge them into cocoa beans and improve other developer's utilities, so you can save time and complexity managing code across multiple projects.
 
+## Relevant Links
+* [Spigot javadocs](https://cocoa-beans.apartium.net/%version%/spigot/)
+* [Common javadocs](https://cocoa-beans.apartium.net/%version%/common/)
+* [Commands javadocs](https://cocoa-beans.apartium.net/%version%/commands/)
```
Comment (liorsl) [inline Writerside/topics/Home.md:9]: Forgot something????

File: .github/workflows/pull_request.yml [hunk 13]
```diff
@@ -1,34 +1,12 @@
 name: Pull request CI
 
 on:
-  pull_request:
-    branches:
-      - main
```
Comment (liorsl) [inline .github/workflows/pull_request.yml:6]: this should still be here though

**Review by liorsl — APPROVED** [review]

**Pull request merged**